java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -7,39 +7,77 @@
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.sun.jna.Pointer;
 import java.io.IOException;
+import java.util.*;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class UnleashEngine {
     private static final String UTF_8 = "UTF-8";
-    private static final String CUSTOM_STRATEGY_RESULTS = "{}";
+    private static final String EMPTY_STRATEGY_RESULTS = "{}";
+
+    private static final Logger log = LoggerFactory.getLogger(UnleashEngine.class);
     private final YggdrasilFFI yggdrasil;
 
     private final ObjectMapper mapper;
+    private final CustomStrategies customStrategies;
 
     public UnleashEngine() {
         this(new YggdrasilFFI());
     }
 
+    public UnleashEngine(List<IStrategy> customStrategies) {
+        this(new YggdrasilFFI(), customStrategies);
+    }
+
     UnleashEngine(YggdrasilFFI yggdrasil) {
+        this(yggdrasil, null);
+    }
+
+    UnleashEngine(YggdrasilFFI yggdrasil, List<IStrategy> customStrategies) {
         this.yggdrasil = yggdrasil;
         this.mapper = new ObjectMapper();
-        mapper.registerModule(new JavaTimeModule());
-        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        this.mapper.registerModule(new JavaTimeModule());
+        this.mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        if (customStrategies != null && !customStrategies.isEmpty()) {
+            List<String> builtInStrategies =
+                    read(yggdrasil.builtInStrategies(), new TypeReference<>() {});
+            this.customStrategies =
+                    new CustomStrategies(
+                            customStrategies.stream()
+                                    .filter(
+                                            strategy -> {
+                                                if (builtInStrategies.contains(
+                                                        strategy.getName())) {
+                                                    log.warn(
+                                                            "Custom strategy {} is already a built-in strategy. Skipping.",
+                                                            strategy.getName());
```

Comment:
This indentation tho. Is this normal? This is what the formatter wants?

Reply:
Yeah, I don't like it either. I'm going to look into more detailed configuration of spotless to make it look better. I just took the defaults for now